CMakeLists.txt

```diff
@@ -0,0 +1,934 @@
+t/helper/test-read-cache.c t/helper/test-read-graph.c t/helper/test-read-midx.c 
+t/helper/test-ref-store.c t/helper/test-regex.c t/helper/test-repository.c 
+t/helper/test-revision-walking.c t/helper/test-run-command.c t/helper/test-scrap-cache-tree.c 
+t/helper/test-serve-v2.c t/helper/test-sha1.c t/helper/test-sha1-array.c t/helper/test-sha256.c 
+t/helper/test-sigchain.c t/helper/test-strcmp-offset.c t/helper/test-string-list.c 
+t/helper/test-submodule-config.c t/helper/test-submodule-nested-repo-config.c t/helper/test-subprocess.c 
+t/helper/test-trace2.c t/helper/test-urlmatch-normalization.c t/helper/test-xml-encode.c 
+t/helper/test-wildmatch.c t/helper/test-windows-named-pipe.c t/helper/test-write-cache.c)
+
+#https://github.com/git-for-windows/git does not have test-advice.c but replaces it with test-cmp.c and test-iconv.c
+if(EXISTS ${CMAKE_SOURCE_DIR}/t/helper/test-advise.c)
+	list(APPEND test_helper_sources t/helper/test-advise.c)
+else()
+	list(APPEND test_helper_sources t/helper/test-cmp.c t/helper/test-iconv.c)
+endif()
+
+add_executable(test-tool ${test_helper_sources})
+target_link_libraries(test-tool common-main)
+
+set_target_properties(test-fake-ssh test-line-buffer test-svn-fe test-tool
+			PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/t/helper)
+
+if(MSVC)
+	set_target_properties(test-fake-ssh test-line-buffer test-svn-fe test-tool
+				PROPERTIES RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_BINARY_DIR}/t/helper)
+	set_target_properties(test-fake-ssh test-line-buffer test-svn-fe test-tool
+				PROPERTIES RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_BINARY_DIR}/t/helper)
+endif()
+
+#wrapper scripts
+set(wrapper_scripts
+git git-upload-pack git-receive-pack git-upload-archive git-shell 
+test-fake-ssh test-line-buffer test-svn-fe test-tool)
+
+foreach(scr ${wrapper_scripts})
+	file(STRINGS ${CMAKE_SOURCE_DIR}/wrap-for-bin.sh content NEWLINE_CONSUME)
+	string(REPLACE "@@BUILD_DIR@@" "${CMAKE_BINARY_DIR}" content "${content}")
+	string(REPLACE "@@PROG@@" "${scr}${EXE}" content "${content}")
+	file(WRITE ${CMAKE_BINARY_DIR}/bin-wrappers/${scr} ${content})
+endforeach()
+
+file(STRINGS ${CMAKE_SOURCE_DIR}/wrap-for-bin.sh content NEWLINE_CONSUME)
+string(REPLACE "@@BUILD_DIR@@" "${CMAKE_BINARY_DIR}" content "${content}")
+string(REPLACE "@@PROG@@" "git-cvsserver" content "${content}")
+file(WRITE ${CMAKE_BINARY_DIR}/bin-wrappers/git-cvsserver ${content})
+
+#GIT-BUILD-OPTIONS
+set(SHELL_PATH /bin/sh)
+set(TEST_SHELL_PATH /bin/sh)
+set(PERL_PATH /usr/bin/perl)
+set(DIFF diff)
+set(PYTHON_PATH /usr/bin/python)
+set(TAR tar)
+set(NO_CURL )
+set(NO_EXPAT )
+set(USE_LIBPCRE1 )
+set(USE_LIBPCRE2 )
+set(NO_LIBPCRE1_JIT )
+set(NO_PERL )
+set(NO_PTHREADS )
+set(NO_PYTHON )
+set(PAGER_ENV "LESS=FRX LV=-c")
+set(DC_SHA1 YesPlease)
+set(NO_GETTEXT )
+set(RUNTIME_PREFIX true)
+set(X ${EXE})
+if(WIN32)
+	set(NO_UNIX_SOCKETS YesPlease)
+else()
+	set(NO_UNIX_SOCKETS )
+endif()
```

Comment:
Logically, those options should probably be configured at the top of the file.

Reply:
okay